# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/nsIWebNavigation.idl
# Commit: ab54092b8afc
# Full Hash: ab54092b8afc8e56dd3f8162344a4ce38ec45999
# Author: Niklas Goegge <ngogge@mozilla.com>
# Date: 2021-04-29 09:26:05
# Regressor Bug: 1708150
# File Overlap Count: 1
# Description:
#   Bug 1708150 - Add user activation flag to reload, goBack and goForward r=ckerschb,Gijs,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110245
# ==============================================================================

diff -r a9a52f01de13 -r ab54092b8afc docshell/base/nsIWebNavigation.idl
--- a/docshell/base/nsIWebNavigation.idl	Wed Apr 28 11:09:38 2021 +0000
+++ b/docshell/base/nsIWebNavigation.idl	Wed Apr 28 11:26:49 2021 +0000
@@ -62,11 +62,15 @@
    *        If no items have user interaction, the function will fall back
    *        to the first session history entry.
    *
+   * @param {boolean} aUserActivation
+   *        Tells goBack that the call was triggered by a user action (e.g.:
+   *        The user clicked the back button).
+   *
    * @throw NS_ERROR_UNEXPECTED
    *        Indicates that the call was unexpected at this time, which implies
    *        that canGoBack is false.
    */
-  void goBack([optional] in boolean aRequireUserInteraction);
+  void goBack([optional] in boolean aRequireUserInteraction, [optional] in boolean aUserActivation);
 
   /**
    * Tells the object to navigate to the next session history item.  When a
@@ -82,20 +86,28 @@
    *        If no items have user interaction, the function will fall back
    *        to the latest session history entry.
    *
+   * @param {boolean} aUserActivation
+   *        Tells goForward that the call was triggered by a user action (e.g.:
+   *        The user clicked the forward button).
+   *
    * @throw NS_ERROR_UNEXPECTED
    *        Indicates that the call was unexpected at this time, which implies
    *        that canGoForward is false.
    */
-  void goForward([optional] in boolean aRequireUserInteraction);
+  void goForward([optional] in boolean aRequireUserInteraction, [optional] in boolean aUserActivation);
 
   /**
    * Tells the object to navigate to the session history item at a given index.
    *
+   * @param {boolean} aUserActivation
+   *        Tells goForward that the call was triggered by a user action (e.g.:
+   *        The user clicked the forward button).
+   *
    * @throw NS_ERROR_UNEXPECTED
    *        Indicates that the call was unexpected at this time, which implies
    *        that session history entry at the given index does not exist.
    */
-  void gotoIndex(in long index);
+  void gotoIndex(in long index, [optional] in boolean aUserActivation);
 
   /****************************************************************************
    * The following flags may be bitwise combined to form the load flags
@@ -252,6 +264,11 @@
   const unsigned long LOAD_FLAGS_BYPASS_LOAD_URI_DELEGATE = 0x4000000;
 
   /**
+   * This load has a user activation. (e.g: reload button was clicked)
+   */
+  const unsigned long LOAD_FLAGS_USER_ACTIVATION = 0x8000000;
+
+  /**
    * Loads a given URI.  This will give priority to loading the requested URI
    * in the object implementing this interface.  If it can't be loaded here
    * however, the URI dispatcher will go through its normal process of content